source/includes/read/retrieve.cpp

```diff
@@ -0,0 +1,47 @@
+#include <cstdint>
+#include <iostream>
+#include <vector>
+
+#include <bsoncxx/builder/basic/document.hpp>
+#include <bsoncxx/json.hpp>
+#include <mongocxx/client.hpp>
+#include <mongocxx/instance.hpp>
+#include <mongocxx/stdx.hpp>
+#include <mongocxx/uri.hpp>
+
+using bsoncxx::builder::basic::kvp;
+using bsoncxx::builder::basic::make_document;
+
+int main() {
+    mongocxx::instance instance{};
+    mongocxx::uri uri("<connection string>");
+    mongocxx::client client(uri);
+
+    // start-db-coll
+    auto db = client["sample_training"];
+    auto collection = db["companies"];
+    // end-db-coll
+
+    // start-find-one
+    auto result = collection.find_one(make_document(kvp("name", "LinkedIn")));
+    std::cout << bsoncxx::to_json(*result) << std::endl;
+    // end-find-one
+
+    // start-find-many
+    auto cursor = collection.find(make_document(kvp("founded_year", 1970)));
+    // end-find-many
+
+    // start-cursor
+    auto cursor = collection.find(make_document(kvp("founded_year", 1970)));
+    for(auto&& doc : cursor) {
+        std::cout << bsoncxx::to_json(doc) << "\n" << std::endl;
```

Comment:
```suggestion
        std::cout << bsoncxx::to_json(doc) << std::endl;
```

Remove duplicate newline: `\n` and `std::endl`.